### ISSUE TITLE
[ENH] Be more intentional/memory conscious with data saved to forecasters such as _y and _X

### ISSUE THREAD

**Issue by hudsonhoch** [sktime/sktime#7416]
**Is your feature request related to a problem? Please describe.**
Forecasters default to saving `_y` and `_X` which can hog memory, especially with high frequency data, without a real need to save all of that information.

**Describe the solution you'd like**
I think ideally `_fit` takes care of what should/should not be saved for downstream use. I'm probably not familiar enough with the ecosystem to comment on the implementation.

There probably needs to be some discussion around what this would mean for `update`.

**Describe alternatives you've considered**
Locally, to get around memory issues I've implemented wrapper classes with `remove_data` methods but I this approach is just a bandaid.

**Additional context**
Discussion on #6914
Specific example of unnecessary saving of `_y` and `_X` #7405

**Comment by fkiraly** [comment]
Agreed, I think it is not a good state that all forecasters store `_y` and `_X`.

I would also agree that this was not such a good design decision - even though it enables `update` for all forecasters, the current implementation forces memory usage on all users, especially in serialization.

I think we need to roll this back and replace by more memory conscious mechanisms, the steps in which I would do this:

1. We should ensure that `fit` and `predict` work with all forecasters if the config `remember_data=False` is set. This will likely fail for a couple forecasters, including those you have identified - the "cheap" way to fix would be, in `_fit` to check whether `_y` and `_X` exist - if not, remember.
    * this test could be added to `TestAllForecasters`
2. Once this is done, we are at a point where we could uniformly switch `remember_data=False` as a default, following the deprecation guide https://www.sktime.net/en/latest/developer_guide/deprecation.html
    * we should think deeply about the transition with clear messaging, and how to give users a clear way to recover the current `update` behaviour. I think the cleanest would be to introduce a wrapper that adds the `update` capability, but it is not clear whether it is the best transition behaviour.

Comments and opinions appreciated! FYI @sktime/core-developers

**Comment by fkiraly** [comment]
I have added such a default test now in a PR:
https://github.com/sktime/sktime/pull/7428

This can be used as a diagnostic, and should flag the `_StatsModelsAdapter` as a failing example, as per #7405.
